newIDE/app/src/UI/DrawerTopBar.js

```diff
@@ -49,15 +51,34 @@ const DrawerTopBar = (props: Props) => {
         elevation={0}
       >
         <Toolbar style={styles.toolbar}>
-          <Tooltip
-            title={props.title}
-            placement="bottom"
-            enterDelay={tooltipEnterDelay}
-          >
-            <Typography variant="h6" style={styles.title}>
-              {props.title}
-            </Typography>
-          </Tooltip>
+          <LineStackLayout noMargin expand alignItems="center">
+            {props.icon && (
+              <IconButton
+                onClick={props.onClose}
+                edge="start"
+                color="inherit"
+                size="small"
+                id={`${props.id}-icon`}
+              >
+                {props.icon}
+              </IconButton>
+            )}
+            {props.title.length > 30 ? (
+              <Tooltip
+                title={props.title}
+                placement="bottom"
+                enterDelay={tooltipEnterDelay}
+              >
+                <Typography variant="h6" style={styles.title}>
+                  {props.title}
```

Comment:
Should the title be truncated here?

Reply:
Or a maxWidth somewhere?

Reply:
I think the style of the title handles it by adding ellipsis automatically (I haven't touched that part, it seems to work)